File: src/DutchMetar.Web.Client/src/app/shared/models/airport-details.ts
import { AirportCurrentMetar } from './airport-current-metar';
import { MeteoCondition } from '../types/meteo-condition';

export type AirportDetails = {
    icao: string;
    name?: string;
    meteoCondition: MeteoCondition;
    lastUpdated: Date;
    latestWeather?: AirportCurrentMetar
}

File: src/DutchMetar.Web.Server/Models/AirportDetails.cs
using DutchMetar.Core.Helpers.Enums;

namespace DutchMetar.Web.Server.Models;

public class AirportDetails
{
    public required string Icao { get; set; }
    
    public MetarMeteoCondition MeteoCondition { get; set; } = MetarMeteoCondition.None;

    public DateTimeOffset LastUpdated { get; set; } = DateTimeOffset.MinValue;
    
    public AirportCurrentMetar? LatestWeather { get; set; }
}